main.go

```diff
@@ -22,8 +23,13 @@ var (
 	collectInterfaceFeatures = flag.Bool("collector.interface-features.enable", true, "Collect interface features")
 	excludeInterfaces        = flag.String("exclude.interfaces", "", "Comma seperated list of interfaces to exclude")
 	includeInterfaces        = flag.String("include.interfaces", "", "Comma seperated list of interfaces to include")
+	excludeInterfacesRegex   = flag.String("exclude.interfaces-regex", "", "Regex of interfaces to exclude")
+	includeInterfacesRegex   = flag.String("include.interfaces-regex", "", "Regex of interfaces to include")
 	excludeInterfacesDown    = flag.Bool("exclude.interfaces-down", false, "Don't report on interfaces being management DOWN")
 	powerUnitdBm             = flag.Bool("collector.optical-power-in-dbm", false, "Report optical powers in dBm instead of mW (default false -> mW)")
+
+	excludeInterfacesRegexCompiled = &regexp.Regexp{}
+	includeInterfacesRegexCompiled = &regexp.Regexp{}
```

Comment:
Those global varaibles are not nice - as you already discovered. 
But: Currently a request does not have any state and it would take some major refactoring to introduce a somewhat non-global state around the request. 

We might do this further down the line, for now, they can stay :smile: